lib/ansible/plugins/action/pause.py

```diff
@@ -169,7 +170,8 @@ def run(self, tmp=None, task_vars=None):
                     if not seconds:
                         new_settings = termios.tcgetattr(fd)
                         new_settings[0] = new_settings[0] | termios.ICRNL
-                        new_settings[3] = new_settings[3] | termios.ICANON
+                        if seconds or 'prompt' in self._task.args:
+                            new_settings[3] = new_settings[3] | termios.ICANON
```

Comment:
if we add ` | termios.ISIG` here, it makes the `^C` respond immediately, though that's aborting the program and not raising our `KeyboardInterrput`. That's the closest thing I can find and still keep the ability to interactively edit the input.